frappe/database/db_manager.py

```diff
@@ -80,12 +80,14 @@ def restore_database(target, source, user, password):
 		if pipe:
 			print('Creating Database...')
 
-		command = '{pipe} mysql -u {user} -p{password} -h{host} {target} {source}'.format(
+		command = '{pipe} mysql -u {user} -p{password} -h{host} ' + ('-P{port}' if frappe.db.port else '') + ' {target} {source}'
+		command = command.format(
 			pipe=pipe,
 			user=esc(user),
 			password=esc(password),
 			host=esc(frappe.db.host),
 			target=esc(target),
-			source=source
+			source=source,
+			port=esc(frappe.db.port)
```

Comment:
we don't need to escape special characters for port if we strictly set it to `int` in the click command

```suggestion
			port=frappe.db.port
```